### PR TITLE
MRG, FIX: make combine_evoked consistent with Matti's manual

### PR DESCRIPTION
**Pull request by drammock**
Per offline discussion w/ @agramfort @larsoner @jasmainak and @jona-sassenhagen. Needs input:

- here when `weights='equal'` the weights are all `1` (previously `1/N`). `combine_tfr` still uses the `1/N` convention; should it be changed too? `mne.grand_average` uses both `combine_evoked` and `combine_tfr` internally, so it might be good if they worked the same way? 
- Regardless of whether we change the behavior of `combine_tfr`, the new behavior of `combine_evoked(..., weights='equal')` is not really consistent with the intuitive meaning of "grand average" (it's really a "grand sum" now). Should `mne.grand_average` be changed to effectively undo this change to `combine_evoked` (by passing pre-computed weights, or by dividing the data afterwards)?

### CODE REVIEW

File: mne/evoked.py [hunk 1]
```diff
@@ -881,14 +881,24 @@ def combine_evoked(all_evoked, weights):
...
+            new_nave = np.sum(naves)
+        else:
+            weights = np.ones_like(naves)
+            new_nave = 1. / np.sum(1. / naves)
+        # for the new_nave calculation, cf. Matti's manual (pp 128-129); both
+        # of the formulae above are equivalent to the general formula below,
+        # but reduce numerical error.
+    else:
+        weights = np.array(weights, float)
+        new_nave = 1. / np.sum(weights ** 2 / naves)
```
Comment (jasmainak) [inline mne/evoked.py]: could you unify these `new_nave` calculations into one formula? or is that not possible?

**Comment by jasmainak** [comment]
Did you check that the examples don't change. E.g., this one: https://www.martinos.org/mne/stable/auto_examples/inverse/plot_compute_mne_inverse_epochs_in_label.html#sphx-glr-auto-examples-inverse-plot-compute-mne-inverse-epochs-in-label-py ? Circle doesn't seem to build them

File: mne/evoked.py (hunk 1 above)
Comment (drammock) [reply mne/evoked.py]: If I use the unified formula, then some of the tests in [test_arithmetic](https://github.com/mne-tools/mne-python/blob/master/mne/tests/test_evoked.py#L461) will fail, because `nave` will end up as, e.g., `30.00000001` instead of `30`.

**Comment by drammock** [comment]
@jasmainak I did not check tutorials/examples. I expect some of them will change if they use `combine_evoked(..., weights='equal')` but maybe not in a way that needs updating.  I can check, and also do a `circle_full` when I push any subsequent changes to make sure.

**Comment by agramfort** [comment]
@drammock how would you handle the changes for the next release? bug fix? deprecation?

**Comment by drammock** [comment]
@agramfort:

1. I think of the changes to `combine_evoked` as a bugfix.
2. the more I ponder it, I think the changes we make here to `combine_evoked` should *not* propogate to `mne.grand_average`, so although we'll need to update the code there to undo the effect of these changes, its behavior will not change (so no need for deprecation cycle there)
3. I haven't looked carefully at `combine_tfr` so I don't know whether a parallel change makes sense there, and if so, whether it would count as a bugfix on similar grounds.

So in summary I lean toward "bugfix" but I'm still waiting for someone who works with TFR regularly to weigh in whether it should also change.

**Comment by agramfort** [comment]
> I think of the changes to combine_evoked as a bugfix.

ok for me. We'll need to put this in the highlights of the next release

> the more I ponder it, I think the changes we make here to combine_evoked should not propogate to mne.grand_average, so although we'll need to update the code there to undo the effect of these changes, its behavior will not change (so no need for deprecation cycle there)

agreed

> I haven't looked carefully at combine_tfr so I don't know whether a parallel change makes sense there, and if so, whether it would count as a bugfix on similar grounds.

I have not thought about this but the structure of the noise for a TFR
object is not Gaussian so the weight
computations are likely to be different or even not apply.

> So in summary I lean toward "bugfix" but I'm still waiting for someone who works with TFR regularly to weigh in whether it should also change.

ok but I think that for TFR the weights with nave strategy make less
sense than with evoked.

**Comment by drammock** [comment]
@agramfort this is ready for review/merge.  I'll push one more commit now to update whats_new

File: doc/whats_new.rst [hunk 2]
```diff
@@ -3256,8 +3260,6 @@ of commits):
 
 .. _Lukas Breuer: http://www.researchgate.net/profile/Lukas_Breuer
 
-.. _Federico Raimondo: https://github.com/fraimondo
```
Comment (agramfort) [inline doc/whats_new.rst]: why did you remove this one?

File: mne/evoked.py [hunk 3]
```diff
@@ -901,24 +912,8 @@ def combine_evoked(all_evoked, weights):
...
-    #
-    # We know that the variance of a weighted sample mean is:
-    #
-    #    σ^2 = w_1^2 σ_1^2 + w_2^2 σ_2^2 + ... + w_n^2 σ_n^2
-    #
-    # We estimate the variance of each evoked instance as 1 / nave to get:
-    #
-    #    σ^2 = w_1^2 / nave_1 + w_2^2 / nave_2 + ... + w_n^2 / nave_n
-    #
-    # And our resulting nave is the reciprocal of this:
```
Comment (agramfort) [inline mne/evoked.py]: can you keep this block of text to explain the weight computation?

File: doc/whats_new.rst (hunk 2 above)
Comment (drammock) [reply doc/whats_new.rst]: Duplicate, It's in there twice

File: mne/evoked.py (hunk 3 above)
Comment (drammock) [reply mne/evoked.py]: Sure I'll add that back in tomorrow morning first thing.

File: mne/evoked.py (hunk 1 above)
Comment (larsoner) [reply mne/evoked.py]: I'd still rather use the unified formula, it makes it clear the computation is the same in all cases, and that's not clear in the code any more. I would change the tests to use `allclose`

Comment (drammock) [reply mne/evoked.py]: OK, I've done this, though I'm fully expecting mailing list complaints saying "why is my `nave` 30.00001 now?"  For the record I'd much rather use 3 different (equivalent) formulae that avoid numerical error when possible, and leave detailed comments in the code showing why they're equivalent.

**Comment by larsoner** [comment]
LGTM +1 for merge once CIs come back happy

File: mne/evoked.py (hunk 1 above)
Comment (jasmainak) [reply mne/evoked.py]: why not use: https://docs.scipy.org/doc/numpy/reference/generated/numpy.round_.html for that then?

File: mne/evoked.py [hunk 4]
```diff
@@ -883,42 +884,45 @@ def combine_evoked(all_evoked, weights):
...
+    #    σ² = w₁² σ₁² + w₂² σ₂² + ... + wₙ² σₙ²
+    #
+    # We estimate the variance of each evoked instance as 1 / nave to get:
+    #
+    #    σ² = w₁² / nave₁ + w₂² / nave₂ + ... + wₙ² / naveₙ
+    #
+    # And our resulting nave is the reciprocal of this:
+    new_nave = 1. / np.sum(weights ** 2 / naves)
+    # This general formula is equivalent to formulae in Matti's manual
+    # (pp 128-129), where:
```
Comment (jasmainak) [inline mne/evoked.py]: sorry late to the party, but Matti's manual is now merged with the online documentation. There is in fact no pdf that you can find from the MNE website. Thus I think it's better to just link to the online documentation.

Comment (drammock) [reply mne/evoked.py]: >  There is in fact no pdf that you can find from the MNE website.

When I log in to [the download page for MNE-C](http://www.nmr.mgh.harvard.edu/martinos/userInfo/data/MNE_register/login.php) I see `MNE-manual-2.7.3.pdf`

File: mne/evoked.py (hunk 1 above)
Comment (drammock) [reply mne/evoked.py]: @jasmainak because we don't *always* want to round, and if you're going to do something special for the cases where numerical error arises, why not do something special that avoids the error in the first place?

File: mne/evoked.py (hunk 4 above)
Comment (jasmainak) [reply mne/evoked.py]: Oh I see. It wasn't obvious to me :)
